diff --git a/include/spdk_internal/log.h b/include/spdk_internal/log.h
index ce96882..a10ebe9 100644
--- a/include/spdk_internal/log.h
+++ b/include/spdk_internal/log.h
@@ -74,8 +74,16 @@ __attribute__((constructor)) static void register_trace_flag_##flag(void) \
 		}										\
 	} while (0)
 
+#define DEBUG
 #ifdef DEBUG
 
+
+#define SPDK_DEBUGLOG(FLAG, ...)								\
+	do {											\
+			spdk_log(SPDK_LOG_INFO, __FILE__, __LINE__, __func__, __VA_ARGS__);	\
+	} while (0)
+
+/*
 #define SPDK_DEBUGLOG(FLAG, ...)								\
 	do {											\
 		extern struct spdk_trace_flag FLAG;						\
@@ -83,7 +91,7 @@ __attribute__((constructor)) static void register_trace_flag_##flag(void) \
 			spdk_log(SPDK_LOG_INFO, __FILE__, __LINE__, __func__, __VA_ARGS__);	\
 		}										\
 	} while (0)
-
+*/
 #define SPDK_TRACEDUMP(FLAG, LABEL, BUF, LEN)						\
 	do {										\
 		extern struct spdk_trace_flag FLAG;					\
diff --git a/lib/bdev/bdev.c b/lib/bdev/bdev.c
index 93b222f..d8312a1 100644
--- a/lib/bdev/bdev.c
+++ b/lib/bdev/bdev.c
@@ -877,14 +877,18 @@ _spdk_bdev_io_submit(void *ctx)
 	struct spdk_io_channel *ch = bdev_ch->channel;
 	struct spdk_bdev_module_channel	*module_ch = bdev_ch->module_ch;
 
+	printf("%s() called \n", __func__);
+
 	bdev_io->submit_tsc = spdk_get_ticks();
 	bdev_ch->io_outstanding++;
 	module_ch->io_outstanding++;
 	bdev_io->in_submit_request = true;
 	if (spdk_likely(bdev_ch->flags == 0)) {
 		if (spdk_likely(TAILQ_EMPTY(&module_ch->nomem_io))) {
+			printf("%s() submit_request \n", __func__);
 			bdev->fn_table->submit_request(ch, bdev_io);
 		} else {
+			printf("%s() insert tail \n", __func__);
 			bdev_ch->io_outstanding--;
 			module_ch->io_outstanding--;
 			TAILQ_INSERT_TAIL(&module_ch->nomem_io, bdev_io, link);
@@ -910,11 +914,15 @@ spdk_bdev_io_submit(struct spdk_bdev_io *bdev_io)
 
 	assert(bdev_io->status == SPDK_BDEV_IO_STATUS_PENDING);
 
+	printf("%s() called \n", __func__);
+
 	if (bdev_io->ch->flags & BDEV_CH_QOS_ENABLED) {
 		bdev_io->io_submit_ch = bdev_io->ch;
 		bdev_io->ch = bdev->qos.ch;
+		printf("%s() send_msg \n", __func__);
 		spdk_thread_send_msg(bdev->qos.thread, _spdk_bdev_io_submit, bdev_io);
 	} else {
+		printf("%s() _io_submit\n", __func__);
 		_spdk_bdev_io_submit(bdev_io);
 	}
 }
@@ -1540,7 +1548,10 @@ spdk_bdev_write(struct spdk_bdev_desc *desc, struct spdk_io_channel *ch,
 {
 	uint64_t offset_blocks, num_blocks;
 
+	printf("%s() called \n", __func__);
+
 	if (spdk_bdev_bytes_to_blocks(desc->bdev, offset, &offset_blocks, nbytes, &num_blocks) != 0) {
+		printf("%s(). spdk_bdev_bytes_to_blocks fails \n", __func__);
 		return -EINVAL;
 	}
 
@@ -1556,6 +1567,8 @@ spdk_bdev_write_blocks(struct spdk_bdev_desc *desc, struct spdk_io_channel *ch,
 	struct spdk_bdev_io *bdev_io;
 	struct spdk_bdev_channel *channel = spdk_io_channel_get_ctx(ch);
 
+	printf("%s() called \n", __func__);
+
 	if (!desc->write) {
 		return -EBADF;
 	}
diff --git a/lib/bdev/nvme/bdev_nvme.c b/lib/bdev/nvme/bdev_nvme.c
index 7995416..344fcae 100644
--- a/lib/bdev/nvme/bdev_nvme.c
+++ b/lib/bdev/nvme/bdev_nvme.c
@@ -191,6 +191,8 @@ bdev_nvme_writev(struct nvme_bdev *nbdev, struct spdk_io_channel *ch,
 {
 	struct nvme_io_channel *nvme_ch = spdk_io_channel_get_ctx(ch);
 
+	printf("%s() called \n", __func__);
+
 	SPDK_DEBUGLOG(SPDK_LOG_BDEV_NVME, "write %lu blocks with offset %#lx\n",
 		      lba_count, lba);
 
@@ -473,6 +475,7 @@ _bdev_nvme_submit_request(struct spdk_io_channel *ch, struct spdk_bdev_io *bdev_
 static void
 bdev_nvme_submit_request(struct spdk_io_channel *ch, struct spdk_bdev_io *bdev_io)
 {
+	printf("%s() called \n", __func__);
 	int rc = _bdev_nvme_submit_request(ch, bdev_io);
 
 	if (spdk_unlikely(rc != 0)) {
@@ -1266,6 +1269,8 @@ bdev_nvme_queue_cmd(struct nvme_bdev *bdev, struct spdk_nvme_qpair *qpair,
 	bio->iovpos = 0;
 	bio->iov_offset = 0;
 
+	printf("%s() called \n", __func__);
+
 	if (direction == BDEV_DISK_READ) {
 		rc = spdk_nvme_ns_cmd_readv(bdev->ns, qpair, lba,
 					    lba_count, bdev_nvme_queued_done, bio, 0,
diff --git a/lib/log/log.c b/lib/log/log.c
index 19b205d..0e58326 100644
--- a/lib/log/log.c
+++ b/lib/log/log.c
@@ -49,6 +49,8 @@ void
 spdk_log_open(void)
 {
 	openlog("spdk", LOG_PID, LOG_LOCAL7);
+
+	printf("[spdk: opened log]\n");
 }
 
 void
@@ -65,6 +67,8 @@ spdk_log(enum spdk_log_level level, const char *file, const int line, const char
 	char buf[MAX_TMPBUF];
 	va_list ap;
 
+	//printf("[spdk: log]\n");
+
 	switch (level) {
 	case SPDK_LOG_ERROR:
 		severity = LOG_ERR;
